File: core/contractsdk/go/example/call/c2/c2.go
package main

import (
	"math/big"
	"strconv"

	"github.com/xuperchain/xuperchain/core/contractsdk/go/code"
	"github.com/xuperchain/xuperchain/core/contractsdk/go/driver"
)

type c2 struct{}

func (c *c2) Initialize(ctx code.Context) code.Response {
	return code.OK(nil)
}

func (c *c2) Invoke(ctx code.Context) code.Response {
	var cnt int
	cntstr, _ := ctx.GetObject([]byte("cnt"))
	if cntstr != nil {
		cnt, _ = strconv.Atoi(string(cntstr))
	}
	cnt += 1000

	args := ctx.Args()
	toaddr := string(args["to"])
	amount := big.NewInt(1000)
	err := ctx.Transfer(toaddr, amount)
	if err != nil {
		return code.Error(err)
	}
	err = ctx.PutObject([]byte("cnt"), []byte(strconv.Itoa(cnt)))
	if err != nil {
		return code.Error(err)
	}

	cntstr = []byte(strconv.Itoa(cnt))
	return code.Response{
		Status:  200,
		Message: string(cntstr),
		Body:    cntstr,
	}
}

func main() {
	driver.Serve(new(c2))
}

File: core/contractsdk/java/example/call/c2/src/main/java/com/baidu/xuper/c2.java
package com.baidu.xuper;

import java.math.BigInteger;

/**
 * c2
 *
 */
public class c2 implements Contract
{
    @Override
    @ContractMethod
    public Response initialize(Context ctx) {
        return Response.ok("initialize success".getBytes());
    }

    @ContractMethod
    public Response invoke(final Context ctx) {
        String cntKey = "cnt";
        BigInteger counter;
        byte[] value = ctx.getObject(cntKey.getBytes());
        if (value == null) {
            counter = BigInteger.valueOf(0);
        } else {
            counter = new BigInteger(value);
        }

        counter = counter.add(BigInteger.valueOf(1000));

        byte[] toByte = ctx.args().get("to");
        if (toByte == null) {
            return Response.error("missing to");
        }
        final String to = new String(toByte);
        ctx.transfer(to,"1000");

        ctx.putObject(cntKey.getBytes(), counter.toByteArray());

        return Response.ok(counter.toByteArray());
    }


    public static void main(String[] args) {
        Driver.serve(new c2());
    }
}
